### PR TITLE
Create YAML pipeline for PHP

### PR DESCRIPTION
**Pull request by MIchaelMainer**
Using same naming format as PowerShell and
Azure SDKs for pipeline folder.

On branch mainer/pipeline/v10generate2
Changes to be committed:
	new file:   .azure-pipelines/generate-v1.0-sdk.yml

### CODE REVIEW

**Comment by MIchaelMainer** [comment]
@ramsessanchez @zengin @peombwa @ddyett 

Can you help this orphaned PR with review?

File: .azure-pipelines/generate-v1.0-sdk.yml [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+
+- task: PowerShell@2
+  displayName: 'Run PHP tests'
+  inputs:
+    targetType: inline
+    workingDirectory: '$(Build.SourcesDirectory)/msgraph-sdk-php'
+    pwsh: true 
+    script: |
+      $installDir = "c:\tools\php"
+      choco install composer --ia "/DEV=$installDir /PHP=$installDir"
```
Comment (MIchaelMainer) [inline .azure-pipelines/generate-v1.0-sdk.yml:130]: This choco install may no longer be necessary per this PR: https://github.com/actions/virtual-environments/pull/588

File: .azure-pipelines/generate-v1.0-sdk.yml [hunk 2]
```diff
@@ -0,0 +1,172 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
```
Comment (MIchaelMainer) [inline .azure-pipelines/generate-v1.0-sdk.yml:3]: nitpick: there should be documentation here describing what this pipeline does.

File: .azure-pipelines/generate-v1.0-sdk.yml [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+name: $(BuildDefinitionName)_$(SourceBranchName)_$(Date:yyyyMMdd)$(Rev:.r)
+
+trigger: none # disable triggers based on commits.
```
Comment (MIchaelMainer) [inline .azure-pipelines/generate-v1.0-sdk.yml:6]: nitpick: Add note that this pipeline is intended to be triggered only after a metadata has been successfully cleaned and captured. It is uses another pipeline as a trigger.

File: .azure-pipelines/generate-v1.0-sdk.yml [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+steps:
+- checkout: msgraph-sdk-php
+  clean: true
+  fetchDepth: 1
+  persistCredentials: true
+- checkout: msgraph-metadata
+  clean: true
+  fetchDepth: 1
+
+- task: PowerShell@2 # Setup environment variables and make them available to all tasks. See [1] for more info.
```
Comment (zengin) [inline .azure-pipelines/generate-v1.0-sdk.yml:43]: NIT: not clear what [1] refers to.

File: .azure-pipelines/generate-v1.0-sdk.yml [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+    ref: master # 
+  pipelines:
+  - pipeline: publishMetadata # This pipeline produces an metadata artifact that we need to validate.
+    source: (v1.0 - 3) msgraph-publish-cleanmetadata
+    trigger: # TODO: validate that this pipeline triggers as expected.
+      branches:
+      - master
+
+pool:
+  vmImage: windows-latest # Info about this image: [0][1]
```
Comment (zengin) [inline .azure-pipelines/generate-v1.0-sdk.yml:29]: NIT: not clear what [0][1] refers to

File: .azure-pipelines/generate-v1.0-sdk.yml [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+  displayName: 'Remove generated files from the repo'
+  inputs:
+    targetType: inline
+    pwsh: true
+    script: |
+      Remove-Item -Recurse $env:REPOMODELSDIR | Write-Host
+      Write-Host "Removed the existing generated files in the repo." -ForegroundColor Green
+
+- task: PowerShell@2
+  displayName: 'Typewriter: generate v1.0 .NET files'
```
Comment (zengin) [inline .azure-pipelines/generate-v1.0-sdk.yml:103]: PHP

File: .azure-pipelines/generate-v1.0-sdk.yml [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+  displayName: 'Git: stage and commit generated files'
+  inputs:
+    targetType: inline
+    pwsh: true
+    workingDirectory: '$(Build.SourcesDirectory)/msgraph-sdk-php'
+    script: |
+      Write-Host "About to add files....." -ForegroundColor Green
+      git add . | Write-Host
+      git commit -m "Update generated files with build $env:BUILD_BUILDID" | Write-Host
+      Write-Host "Added and commited generated files." -ForegroundColor Green
```
Comment (zengin) [inline .azure-pipelines/generate-v1.0-sdk.yml:144]: NIT: committed

File: .azure-pipelines/generate-v1.0-sdk.yml (hunk 5 above)
Comment (zengin) [reply .azure-pipelines/generate-v1.0-sdk.yml:29]: my bad.

**Pull request merged**